gtsfm/common/gtsfm_data.py

```diff
@@ -226,7 +227,7 @@ def from_selected_cameras(cls, gtsfm_data: "GtsfmData", camera_indices: List[int
         Returns:
             New object with the selected cameras and associated tracks.
         """
-        new_data = cls(number_images=len(camera_indices))
+        new_data = cls(number_images=gtsfm_data.number_images())
```

Comment:
this reverts a change from a few weeks back. we cannot change `number_images()` until the final I/O dump, because `GtsfmData.get_camera_poses()` uses `number_images()` to make a list of estimated poses, and that list needs to have the same length as the list of GT poses.